plugins/woocommerce/client/admin/client/settings/index.js

```diff
@@ -19,9 +19,45 @@ import './settings.scss';
 
 const node = document.getElementById( 'wc-settings-page' );
 
+const appendSettingsScripts = ( scripts ) => {
+	return scripts.map( ( script ) => {
+		const scriptElement = document.createElement( 'script' );
+		scriptElement.src = script;
+		scriptElement.onerror = () => {
+			// eslint-disable-next-line no-console
+			console.error( `Failed to load script: ${ script }` );
+		};
+		document.body.appendChild( scriptElement );
+		return scriptElement;
+	} );
+};
+
+const removeSettingsScripts = ( scripts ) => {
+	scripts.forEach( ( script ) => {
+		document.body.removeChild( script );
+	} );
+};
+
+const SETTINGS_SCRIPTS = window.wcSettings?.admin?.settingsScripts || [];
+
 const Settings = () => {
 	const { activePage, activeSection } = useActiveRoute();
 
+	useLayoutEffect( () => {
+		const scripts = Array.from(
+			new Set( [
+				...( SETTINGS_SCRIPTS._default || [] ),
+				...( SETTINGS_SCRIPTS[ activePage ] || [] ),
+			] )
+		);
+
+		const scriptsElements = appendSettingsScripts( scripts );
+
+		return () => {
+			removeSettingsScripts( scriptsElements );
+		};
+	}, [ activePage, activeSection ] );
```

Comment:
`activeSection` is not used in the effect, is this intentional?

Reply:
> activeSection is not used in the effect, is this intentional?

It's intentional. We need to rerun the scripts when the section changes as well.